data/core/commands/doc.lua

```diff
@@ -93,11 +93,14 @@ local function cut_or_copy(delete)
   system.set_clipboard(full_text)
 end
 
-local function split_cursor(direction)
+local function split_cursor(dv, direction)
   local new_cursors = {}
+  local dv_translate = direction < 0
+    and DocView.translate.previous_line
+    or DocView.translate.next_line
   for _, line1, col1 in doc():get_selections() do
     if line1 + direction >= 1 and line1 + direction <= #doc().lines then
```

Comment:
At this point let's also use the argument here instead of `doc`.
```suggestion
  for _, line1, col1 in dv:get_selections() do
    if line1 + direction >= 1 and line1 + direction <= #dv.lines then
```

Reply:
Yep

Reply:
It's fine, this will get squashed anyways.